src/string.rs

```diff
@@ -939,3 +949,57 @@ pub extern "C" fn free_rust_external_onebyte(s: *mut char, len: usize) {
     drop(Box::from_raw(slice));
   }
 }
+
+#[derive(Debug, PartialEq)]
+pub enum ValueViewData<'s> {
+  OneByte(&'s [u8]),
+  TwoByte(&'s [u16]),
+}
+
+/// Returns a view onto a string's contents.
+///
+/// WARNING: This does not copy the string's contents, and will therefore be
+/// invalidated if the GC can move the string while the ValueView is alive. It
+/// is therefore required that no GC or allocation can happen while there is an
+/// active ValueView. This requirement may be relaxed in the future.
+///
+/// V8 strings are either encoded as one-byte or two-bytes per character.
+#[repr(C)]
+pub struct ValueView([u8; crate::binding::RUST_v8__String__ValueView_SIZE]);
+
+impl ValueView {
+  #[inline(always)]
+  pub fn new<'s>(
+    scope: &HandleScope<'s, ()>,
+    string: Local<'s, String>,
+  ) -> Self {
+    let mut v = std::mem::MaybeUninit::uninit();
+    unsafe {
+      v8__String__ValueView__CONSTRUCT(
+        v.as_mut_ptr(),
+        scope.get_isolate_ptr(),
+        &*string,
+      );
+      v.assume_init()
+    }
+  }
+
+  #[inline(always)]
+  pub fn data(&self) -> ValueViewData {
```

Comment:
This should be `unsafe`, it's possible to take a 'static reference to string contents (UB)

```rust
let view = ValueView::new(scope, local);
let data: ValueViewData<'static> = view.data(); 
```